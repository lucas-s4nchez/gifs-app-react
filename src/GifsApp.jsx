import React from "react";

export const GifsApp = () => {
  return <div>GifsApp</div>;
};
