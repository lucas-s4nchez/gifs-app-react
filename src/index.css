* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
  box-sizing: border-box;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}
.title {
  font-size: 24px;
}
.title span {
  font-weight: 700;
}
.container {
  max-width: 1200px;
  width: 95%;
  margin: 0 auto;
}
.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 20px;
}

.card {
  position: relative;
  align-content: center;
  align-items: center;
  background-color: #000;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 300px;
  width: 300px;
  justify-content: center;
  overflow: hidden;
}

.card img {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
}
/*Spinner*/
.spinner-container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-block: 30px;
}
.lds-dual-ring {
  display: inline-block;
  width: 80px;
  height: 80px;
}
.lds-dual-ring:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid rgb(25, 0, 255);
  border-color: rgb(25, 0, 255) transparent rgb(25, 0, 255) transparent;
  animation: lds-dual-ring 1.2s linear infinite;
}
@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
