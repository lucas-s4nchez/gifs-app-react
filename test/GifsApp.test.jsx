import { fireEvent, render, screen } from "@testing-library/react";
import { GifsApp } from "../src/GifsApp";

describe("Pruebas en <GifsApp/>", () => {
  const result = "Pizza";

  test("Debe de hacer match con el snapshoot", () => {
    const { container } = render(<GifsApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe arrojar un resultado de busqueda al hacer submit", () => {
    render(<GifsApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByTestId("form");

    fireEvent.input(input, { target: { value: result } });
    fireEvent.submit(form);

    expect(screen.getByText(result)).toBeTruthy();
  });
});
