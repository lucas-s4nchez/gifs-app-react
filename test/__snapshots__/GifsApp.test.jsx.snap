// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifsApp/> Debe de hacer match con el snapshoot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="my-3 text-center"
    >
      GifsApp
    </h1>
    <form
      class="d-flex"
      data-testid="form"
    >
      <input
        class="form-control"
        name="search"
        placeholder="Buscar Gif..."
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
